urbanstats-persistent-data/urbanstats_persistent_data/juxtastat_stats.py

```diff
@@ -46,14 +52,57 @@ def table():
     return conn, c
 
 
-def register_user(user, domain):
+def register_user(user, secure_id, domain):
+    """
+    Register a user with a secure id and domain. First checks
+    if the user is already registered, if so checks if the secure id is correct.
+    If the secure id is incorrect, returns True. Otherwise, updates the domain
+    and secure id and returns False.
+
+    This is Trust on First Use (TOFU) authentication.
+    """
     user = int(user, 16)
+    secure_id = int(secure_id, 16)
     conn, c = table()
+    # check if secure id is already in the database
+    c.execute(
+        "SELECT secure_id FROM JuxtaStatUserSecureID WHERE user=?",
+        (user,),
+    )
+    res = c.fetchone()
+    if res is None:
+        # trust on first use
+        c.execute(
+            "INSERT INTO JuxtaStatUserSecureID VALUES (?, ?)",
+            (user, secure_id),
+        )
+    else:
+        # Ensure that the secure id is correct
+        if res[0] != secure_id:
+            # Security error! Return True to indicate that the user is not authenticated
+            return True
```

Comment:
Canonical way to do this is to return 401, rather than this confusing and arbitrary true/false business

Reply:
See https://github.com/kavigupta/urbanstats/issues/713